services/gitdiff/gitdiff.go

```diff
@@ -441,91 +441,249 @@ func (diff *Diff) LoadComments(issue *models.Issue, currentUser *models.User) er
 
 const cmdDiffHead = "diff --git "
 
-// ParsePatch builds a Diff object from a io.Reader and some
-// parameters.
-// TODO: move this function to gogits/git-module
+// ParsePatch builds a Diff object from a io.Reader and some parameters.
 func ParsePatch(maxLines, maxLineCharacters, maxFiles int, reader io.Reader) (*Diff, error) {
-	var (
-		diff       = &Diff{Files: make([]*DiffFile, 0)}
-		curFile    = &DiffFile{}
-		curSection = &DiffSection{
-			Lines: make([]*DiffLine, 0, 10),
+	var curFile *DiffFile
+
+	diff := &Diff{Files: make([]*DiffFile, 0)}
+
+	sb := strings.Builder{}
+
+	// OK let's set a reasonable buffer size.
+	// This should be let's say at least the size of maxLineCharacters or 4096 whichever is larger.
+	readerSize := maxLineCharacters
+	if readerSize < 4096 {
+		readerSize = 4096
+	}
+
+	input := bufio.NewReaderSize(reader, readerSize)
+	line, err := input.ReadString('\n')
+	if err != nil {
+		return diff, err
+	}
+parsingLoop:
+	for {
+		if err != nil {
```

Comment:
I've attempted to trace through this loop to search for an err that is previously unhandled such that this conditional would be needed. Although it is an unreasonable hour for me to be awake, let alone doing PR reviews, so I am likely missing something, and should return in the morning with fresh eyes.

Reply:
Sorry that's a hang over from earlier code and error handing. I've removed it now and should have improved handling in general.